3-enrich/emr-etl-runner/lib/snowplow-emr-etl-runner/generator.rb

```diff
@@ -0,0 +1,84 @@
+# Copyright (c) 2012-2017 Snowplow Analytics Ltd. All rights reserved.
+#
+# This program is licensed to you under the Apache License Version 2.0,
+# and you may not use this file except in compliance with the Apache License Version 2.0.
+# You may obtain a copy of the Apache License Version 2.0 at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the Apache License Version 2.0 is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the Apache License Version 2.0 for the specific language governing permissions and limitations there under.
+
+# Author::    Ben Fradet (mailto:support@snowplowanalytics.com)
+# Copyright:: Copyright (c) 2012-2017 Snowplow Analytics Ltd
+# License::   Apache License Version 2.0
+
+require 'contracts'
+require 'open-uri'
+require 'avro'
+require 'json'
+
+# Abstract class defining a generator behavior.
+# See implementing classes: EmrClusterGenerator and PlaybookGenerator.
+module Snowplow
+  module EmrEtlRunner
+    module Generator
+
+      include Contracts
+
+      # Print out the Avro record given by create_record to a file using the schema provided by
+      # get_schema
+      Contract ConfigHash, String, String, Bool, Maybe[String], String, ArrayOf[String] => nil
+      def generate(config, version, filename, debug=false, resume_from=nil, resolver='', enrichments=[])
+        raw_schema = get_schema(version)
+        avro_schema = Avro::Schema.parse(raw_schema)
+        datum = create_datum(config, debug, resume_from, resolver, enrichments)
+        if Avro::Schema.validate(avro_schema, datum)
+          json = {
+            "schema" => "iglu:" + get_schema_name_from_version(version),
+            "data" => datum
+          }.to_json
+          File.open(filename, 'w+') { |file|
+            file.write(json)
+          }
+        else
+          raise ConfigError, "Config could not be validated against the schema #{avro_schema}"
+        end
+        nil
+      end
+
+      # Get the associated Avro schema
+      Contract String => String
+      def get_schema(version)
+        url = "https://raw.githubusercontent.com/snowplow/iglu-central/master/schemas/" +
```

Comment:
Yup I'll do that, I went this route not to have to supply a resolver file to generate an emr-config file but it doesn't really seem worth it.